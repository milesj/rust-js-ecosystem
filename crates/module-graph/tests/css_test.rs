mod utils;

use starbase_sandbox::{assert_snapshot, create_sandbox};
use utils::generate_graph_for_file;

mod css {
    use super::*;

    #[test]
    fn normal() {
        let sandbox = create_sandbox("css");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "styles.css"));
    }

    #[test]
    fn modules() {
        let sandbox = create_sandbox("css");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "styles.module.css"));
    }
}
