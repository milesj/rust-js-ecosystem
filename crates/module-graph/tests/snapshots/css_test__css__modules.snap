---
source: crates/module-graph/tests/css_test.rs
expression: "generate_graph_for_file(sandbox.path(), \"styles.module.css\")"
---
{
    2: Module {
        exports: [
            Export {
                kind: Native,
                module_id: None,
                source: None,
                span: None,
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "clearfix",
                    },
                ],
                type_only: false,
            },
            Export {
                kind: Native,
                module_id: None,
                source: None,
                span: None,
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "reset",
                    },
                ],
                type_only: false,
            },
        ],
        fragment: None,
        id: 2,
        imports: [],
        package_name: None,
        path: "/private/root/utils.module.css",
        query: None,
        source: CssModule {
            exports: {
                "clearfix": "_2wbkJG_clearfix",
                "reset": "_2wbkJG_reset",
            },
            file_name: "utils.module.css",
            source: ".clearfix {\n    clear: both;\n}\n\n.reset {\n    margin: 0;\n    padding: 0;\n}\n",
        },
    },
    1: Module {
        exports: [
            Export {
                kind: Native,
                module_id: None,
                source: None,
                span: None,
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "button",
                    },
                ],
                type_only: false,
            },
            Export {
                kind: Native,
                module_id: None,
                source: None,
                span: None,
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "compose-external",
                    },
                ],
                type_only: false,
            },
            Export {
                kind: Native,
                module_id: None,
                source: None,
                span: None,
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "compose-global",
                    },
                ],
                type_only: false,
            },
            Export {
                kind: Native,
                module_id: None,
                source: None,
                span: None,
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "compose-local",
                    },
                ],
                type_only: false,
            },
            Export {
                kind: Native,
                module_id: None,
                source: None,
                span: None,
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "container",
                    },
                ],
                type_only: false,
            },
        ],
        fragment: None,
        id: 1,
        imports: [
            Import {
                kind: SyncStatic,
                module_id: 2,
                source_request: "./utils.module.css",
                span: Span {
                    start: 0,
                    end: 0,
                },
                symbols: [
                    ImportedSymbol {
                        kind: Value,
                        source_name: None,
                        symbol_id: None,
                        name: "clearfix",
                    },
                    ImportedSymbol {
                        kind: Value,
                        source_name: None,
                        symbol_id: None,
                        name: "reset",
                    },
                ],
                type_only: false,
            },
        ],
        package_name: None,
        path: "/root/styles.module.css",
        query: None,
        source: CssModule {
            exports: {
                "button": "EcQGha_button",
                "compose-external": "EcQGha_compose-external",
                "compose-global": "EcQGha_compose-global",
                "compose-local": "EcQGha_compose-local",
                "container": "EcQGha_container",
            },
            file_name: "styles.module.css",
            source: ".container {\n    display: flex;\n}\n\n.button {\n    background-color: black;\n}\n\n.button:hover {\n    background-color: white;\n}\n\n.compose-local {\n    composes: container;\n}\n\n.compose-global {\n    composes: global-class from global;\n}\n\n.compose-external {\n    composes: clearfix reset from \"./utils.module.css\";\n}\n",
        },
    },
}
